### PR TITLE
fix: [CDS-34551]: exporting deployservicestep for gitops

### PR DESCRIPTION
**Pull request by cjlee01**
##### Summary:

exporting deployservicestep for gitops to be used in the new app flow

##### Jira Links:

https://harness.atlassian.net/browse/CDS-34551

##### Screenshots:

image

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by cjlee01** [comment]
retrigger titlecheck

**Review by sahithikolichala — DISMISSED** [review]

**Comment by cjlee01** [comment]
retrigger jest

File: src/microfrontends/ChildAppMounter.tsx [hunk 1]
```diff
@@ -79,6 +80,7 @@ export class ChildAppMounter extends React.Component<ChildAppMounterProps, Child
             YAMLBuilder,
             MonacoDiffEditor
           }}
+          customComponents={{ DeployServiceWidget, DeployEnvironmentWidget }}
```
Comment (vkbansal) [inline src/microfrontends/ChildAppMounter.tsx:83]: Please add it here https://github.com/harness/harness-core-ui/blob/a7d2c91d25038f66adc06a4ba7b920d28027ee64/src/modules/75-cd/RouteDestinations.tsx#L416

**Review by chiragp1212 — APPROVED** [review]

File: src/microfrontends/index.ts [hunk 2]
```diff
@@ -38,6 +40,8 @@ export interface CommonComponents {
     ExtendedMonacoDiffEditorProps & React.RefAttributes<MonacoDiffEditor>
   >
 }
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type CustomComponents = Record<string, React.ComponentType<any>>
```
Comment (abhinavrastogi-harness) [inline src/microfrontends/index.ts:44]: why are we wrapping the components in "CustomComponents" with forced `any`, which disables typechecking?
why cant we pass the required components directly?

**Review by abhinavrastogi-harness — CHANGES_REQUESTED** [review]

File: src/microfrontends/index.ts (hunk 2 above)
Comment (vkbansal) [reply src/microfrontends/index.ts:44]: These custom components can be chosen by the MFE as per their requirements and not required by everyone. `any` is required to make it generic record.

Comment (vkbansal) [reply src/microfrontends/index.ts:44]: Example: https://github.com/wings-software/gitops/pull/615

File: src/microfrontends/ChildAppMounter.tsx [hunk 3]
```diff
@@ -88,4 +95,13 @@ export class ChildAppMounter extends React.Component<ChildAppMounterProps, Child
   }
 }
 
-export default withRouter(ChildAppMounter)
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const ChildAppMounterWithRouter = withRouter(ChildAppMounter as any)
```
Comment (abhinavrastogi-harness) [inline src/microfrontends/ChildAppMounter.tsx:99]: why do we need `any` here? it was working without it earlier

**Review by abhinavrastogi-harness — APPROVED** [review]

File: src/microfrontends/ChildAppMounter.tsx (hunk 3 above)
Comment (vkbansal) [reply src/microfrontends/ChildAppMounter.tsx:99]: After the refactor it's not and I'm unable to figure this out.

**Review by vkbansal — APPROVED** [review]

**Pull request merged**